api/types/plugin.go

```diff
@@ -42,6 +42,10 @@ type PluginConfig struct {
 	// Required: true
 	Description string `json:"Description"`
 
+	// Docker Version used to create the plugin
+	// Required: true
+	DockerVersion string `json:"DockerVersion"`
```

Comment:
I'm thinking of a case where users wants an inventory of all plugins versions. If we have an empty string, they could use `docker plugin inspect <id> --format {{.Config.DockerVersion}}` and if there are ones returning empty know that they were built on 17.03/17.04. If we omit it from the json, then `plugin inspect` will error out.  This makes me lean more towards having an empty version, rather than omitting it. WDYT?